synapse/rest/media/thumbnail_resource.py

```diff
@@ -208,8 +223,13 @@ async def _select_or_generate_remote_thumbnail(
         desired_height: int,
         desired_method: str,
         desired_type: str,
+        max_timeout_ms: int,
     ) -> None:
-        media_info = await self.media_repo.get_remote_media_info(server_name, media_id)
+        media_info = await self.media_repo.get_remote_media_info(
+            server_name, media_id, max_timeout_ms
+        )
+        if not media_info:
+            return
```

Comment:
Related to the `_select_or_generate_local_thumbnail` behavior mentioned above

```suggestion
            respond_404(request)
            return
```

Also applies to `_respond_remote_thumbnail` below